### ISSUE TITLE
Detect when a user might be experience a webcompat issue and help turn off Shields

### ISSUE THREAD

**Issue by karenkliu** [brave/brave-browser#4247]
## Description
Currently, users in the general audience attribute webcompat issues to the browser or website not working correctly and don't identify it as a Shields setting issue. Help the user fix webcompat issues on their own by detecting when they are possibly experiencing a webcompat issue and surfacing instructions to fix it.

## Designs
The design should aim to improve:
Poor onboarding and discovery
Supporting research: A lot of users thought the logo was branding before we added numbers
Education around web incompatibility issues
Education around privacy/security terminology
Fixing web incompatibility issues
Non-expert users should be able to fix incompatibility themselves
Use case: As a subscriber to a Wordpress blogger, I want to know how to view their content after the blogger added an incompatible widget, either by playing around with Shields myself or being instructed to by the blogger.
Expert users should have more granular control to fix issues 
Sending feedback on web incompatibility issues
User understanding of global controls vs. site-specific controls
User quote: “It took me a little while to figure out shields are customized on each domain. Thought it’s for every website I’m visiting. Didn’t know I could turn it off for a single domain.”
Cross-platform consistency
Localization efforts with translations
Enjoyability of using the Brave browser with:
Brand moments
Easter egg interactions
More stats on # of items Shields protected users from

**Comment by karenkliu** [comment]
@cezaraugusto Is this technically feasible?

**Comment by bsclifton** [comment]
@karenkliu I know @too4words and others have been looking at webcompat. I think we'd need to lean on the work they're doing, as there isn't a clear indicator from the browsers perspective on whether it's working correctly or not. 

There are folks doing analysis for shields up/down to find differences which may be webcompat problems

**Comment by karenkliu** [comment]
@bsclifton Yep - this is more of a stopgap solution while they are working on getting a fully automated webcompat solution up and running. We are hoping that in the future the user does not need to interact with the Shields panel at all, because Shields is smart enough to have the right settings per the work the research team is doing. However, in the meantime we still need to provide the user with *something* because the worst case scenario is that they assume Brave Browser is glitchy and switch right back to Chrome. Right now the small percentage of self-selected users who reach out to Brave support are typically met with a first line instruction of "have you tried turning it off?". The design can have use this temporary solution to take care of those cases and help retention for new users.

**Comment by bsclifton** [comment]
cc: @snyderp

**Comment by snyderp** [comment]
I think this is a great idea.  Would also be good to incorporate it with the "report this site is broken to brave" feature that @antonok-edm and @alexwykoff are working on

**Comment by karenkliu** [comment]
@snyderp That's the idea! They are working on the MVP version of what is to become https://github.com/brave/brave-browser/issues/4262; this future design details out what the "report this site is broken to brave" flow looks like integrated into the Shields panel.
